infra/utils.py

```diff
@@ -1,12 +1,32 @@
 """Utility functions."""
-
-from typing import Iterable, List, Sequence, Tuple, TypeVar, Union
+import types
+from typing import Iterable, List, Sequence, Tuple, TypeVar, Union, Type, Any, Callable, cast, Collection
 
 T = TypeVar("T")
 Rotatable = Union[Sequence[T], str]
 Indexable = Union[List[T], Union[T], str]
 
 
+def chunks(items: Collection[T], n: int, chunk_type: Type[Collection[T]] = None) -> Iterable[Collection[T]]:
```

Comment:
Please remove the `chunk_type` parameter - it makes things complicated and people can do stuff like `[set(x) for x in chunked(...)] anyways`. It also adds stuff to the functionality that's beyond splitting a sequence into chunks.

Also, to feel the same as `sorted`, please call it `chunked`.